pkg/autoscaler/autoscaler.go

```diff
@@ -877,4 +895,130 @@ var _ = Describe("Autoscaler should", framework.LabelAutoscaler, Serial, func()
+		It("high priority machineset should be scaled up first [Slow]", func() {
+			By("Creating 2 MachineSets each with 1 replica")
+			var transientMachineSets [2]*machinev1.MachineSet
+			targetedNodeLabel := fmt.Sprintf("%v-priority-expander", autoscalerWorkerNodeRoleLabel)
+			for i := range transientMachineSets {
+				machineSetParams := framework.BuildMachineSetParams(ctx, client, 1)
+				machineSetParams.Labels[targetedNodeLabel] = ""
+				machineSet, err := framework.CreateMachineSet(client, machineSetParams)
+				Expect(err).ToNot(HaveOccurred(), "Failed to create MachineSet %d of %d", i, len(transientMachineSets))
+				cleanupObjects[machineSet.GetName()] = machineSet
+				transientMachineSets[i] = machineSet
+			}
+
+			By("Waiting for all Machines in MachineSets to enter Running phase")
+			framework.WaitForMachineSet(ctx, client, transientMachineSets[0].GetName())
+			framework.WaitForMachineSet(ctx, client, transientMachineSets[1].GetName())
+
+			maxMachineSetReplicas := int32(2)
+			for _, machineSet := range transientMachineSets {
+				By(fmt.Sprintf("Creating a MachineAutoscaler backed by MachineSet %s - min: 1, max: %d",
+					machineSet.GetName(), maxMachineSetReplicas))
+				asr := machineAutoscalerResource(machineSet, 1, maxMachineSetReplicas)
+				Expect(client.Create(ctx, asr)).Should(Succeed(), "Failed to create MachineAutoscaler with min 1/max %d replicas", maxMachineSetReplicas)
+				cleanupObjects[asr.GetName()] = asr
+			}
+
+			By("Create cluster-autoscaler-priority-expander")
+			priorities := "10: \n  - .*" + transientMachineSets[0].GetName() + ".*\n20:\n  - .*" + transientMachineSets[1].GetName() + ".*"
+			cm := configMapResource(priorities)
+			Expect(client.Create(ctx, cm)).Should(Succeed(), "Failed to create ConfigMap with priorities %s", priorities)
+			cleanupObjects[cm.GetName()] = cm
+
+			jobReplicas := int32(4)
+			uniqueJobName := fmt.Sprintf("%s-priority-expander", workloadJobName)
+			By(fmt.Sprintf("Creating scale-out workload %s: jobs: %v, memory: %s",
+				uniqueJobName, jobReplicas, workloadMemRequest.String()))
+			workload := framework.NewWorkLoad(jobReplicas, workloadMemRequest, uniqueJobName, autoscalingTestLabel, "", corev1.NodeSelectorRequirement{
+				Key:      targetedNodeLabel,
+				Operator: corev1.NodeSelectorOpExists,
+			})
+			cleanupObjects[workload.GetName()] = workload
+			Expect(client.Create(ctx, workload)).Should(Succeed(), "Failed to create scale-out workload %s", uniqueJobName)
+
+			By("Check machinesets scale up to 2 machines")
+			expectedReplicas := int32(2)
+			Eventually(func() bool {
+				ms, err := framework.GetMachineSet(ctx, client, transientMachineSets[0].GetName())
+				Expect(err).ToNot(HaveOccurred(), "Failed to get MachineSets %s", transientMachineSets[0].GetName())
+				ms1, err := framework.GetMachineSet(ctx, client, transientMachineSets[1].GetName())
+				Expect(err).ToNot(HaveOccurred(), "Failed to get MachineSets %s", transientMachineSets[1].GetName())
```

Comment:
Should use local gomega
```suggestion
			Eventually(func(g Gomega) bool {
				ms, err := framework.GetMachineSet(ctx, client, transientMachineSets[0].GetName())
				g.Expect(err).ToNot(HaveOccurred(), "Failed to get MachineSets %s", transientMachineSets[0].GetName())
				ms1, err := framework.GetMachineSet(ctx, client, transientMachineSets[1].GetName())
				g.Expect(err).ToNot(HaveOccurred(), "Failed to get MachineSets %s", transientMachineSets[1].GetName())
```
Or better, check them separately and use gomega to assert the error is not present, and match the integer using gomega matchers
```
Eventually(func() (*int, error) {
	ms, err := framework.GetMachineSet(ctx, client, transientMachineSets[0].GetName())
	return ms.Spec.Replicas, err
}).To(HaveValue(Equal(expectedReplicas)), "MachineSet ... should match expected replicas"
```